lighthouse-core/gather/driver.js

```diff
@@ -688,6 +688,20 @@ class Driver {
   }
 
   /**
+   * Returns the flattened list of all DOM nodes within the document.
+   * @param {boolean=} pierce Whether to pierce through shadow trees and iframes.
+   *     True by default.
+   * @return {!Promise<!Array<!Element>>} The found elements, or [], resolved in a promise
+   */
+  getAllNodesInDocument(pierce = true) {
```

Comment:
you filter on nodetype and then return our element instance.
and this pierces iframes whereas the dom helper doesnt.

`getAllElementsInDocumentAndFrames` ?